### PR TITLE
WIP: refactoring User

### PR DESCRIPTION
**Pull request by igkostyuk**
draft request

### CODE REVIEW

File: internal/usecases/user/usecase.go [hunk 1]
```diff
@@ -34,37 +34,22 @@ type Usecases struct {
...
-func (uc *Usecases) Create(ctx context.Context, nu entity.NewUser) (string, error) {
-	hash, err := bcrypt.GenerateFromPassword([]byte(nu.Password), bcrypt.DefaultCost)
+func (uc *Usecases) Create(ctx context.Context, u *entity.User) error {
+	hash, err := bcrypt.GenerateFromPassword([]byte(u.PasswordHash), bcrypt.DefaultCost)
 	if err != nil {
 		return "", fmt.Errorf("generate password hash:%w", err)
 	}
 
-	id := uuid.New().String()
+	u.ID = uuid.New().String()
```
Comment (glbter) [inline internal/usecases/user/usecase.go:43]: db would not generate uuid anumore?

File: internal/usecases/user/usecase.go [hunk 2]
```diff
@@ -34,37 +34,22 @@ type Usecases struct {
 }
 
 // Create Add new user
-func (uc *Usecases) Create(ctx context.Context, nu entity.NewUser) (string, error) {
-	hash, err := bcrypt.GenerateFromPassword([]byte(nu.Password), bcrypt.DefaultCost)
+func (uc *Usecases) Create(ctx context.Context, u *entity.User) error {
+	hash, err := bcrypt.GenerateFromPassword([]byte(u.PasswordHash), bcrypt.DefaultCost)
```
Comment (glbter) [inline internal/usecases/user/usecase.go:38]: generate hasp from user.passwordHash - is not straitforward to understand)))

File: internal/usecases/user/usecase.go [hunk 3]
```diff
@@ -34,37 +34,22 @@ type Usecases struct {
...
-	}
-	u := entity.User{
-		ID:           nu.ID,
-		Name:         nu.Name,
-		Email:        nu.Email,
-		PasswordHash: hash,
-	}
-	return u, uc.repo.Update(ctx, &u)
+func (uc *Usecases) Update(ctx context.Context, u *entity.User) error {
+	return uc.repo.Update(ctx, u)
```
Comment (glbter) [inline internal/usecases/user/usecase.go:52]: probably, we should add entities for each level  in future

File: internal/server/http/auth/handlers.go [hunk 4]
```diff
@@ -74,32 +74,45 @@ func (h *Handlers) CheckPasswordCode(w http.ResponseWriter, r *http.Request) {
 	h.render(w, tk)
 }
 
-func (h *Handlers) ChangePassword(w http.ResponseWriter, r *http.Request) {
-	var req entity.UserNewPassword
-	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
-		h.writeErrorResponse(http.StatusBadRequest, "could not parse request", w)
+// GetToken by basic auth.
+func (h *Handlers) GetToken(w http.ResponseWriter, r *http.Request) {
+	var newUser entity.NewUser
```
Comment (glbter) [inline internal/server/http/auth/handlers.go:79]: as I understood, you moved login and logout to auth handlers, and change password to user. It is ok. 
However, we should think, how we would divide our db into several smaller dbs. If it is going to touch user, we should move it again

Comment (glbter) [reply internal/server/http/auth/handlers.go:79]: it is for further research))

File: internal/entity/user.go [hunk 5]
```diff
@@ -8,27 +8,27 @@ type User struct {
...
+// }
 
-type PasswordsRequest struct {
-	UserID          string `json:"id"`
-	Password        string `json:"password"`
-	PasswordConfirm string `json:"password_confirm"`
-}
+// type PasswordsRequest struct {
+// 	UserID          string `json:"id"`
+// 	Password        string `json:"password"`
```
Comment (glbter) [inline internal/entity/user.go:25]: oh, I used it for my functions, do you really need to delete it?

**Review by glbter — CHANGES_REQUESTED** [review]
you swapped functions without imports, as you told. It is ok in general. I hope, you could fix it later

File: internal/usecases/user/usecase.go (hunk 1 above)
Comment (igkostyuk) [reply internal/usecases/user/usecase.go:43]: hm, it can

File: internal/usecases/user/usecase.go (hunk 2 above)
Comment (igkostyuk) [reply internal/usecases/user/usecase.go:38]: any suggestion ?

File: internal/usecases/user/usecase.go (hunk 3 above)
Comment (igkostyuk) [reply internal/usecases/user/usecase.go:52]: hm ?

File: internal/server/http/auth/handlers.go (hunk 4 above)
Comment (igkostyuk) [reply internal/server/http/auth/handlers.go:79]: any suggestion ?

File: internal/usecases/user/usecase.go (hunk 3 above)
Comment (glbter) [reply internal/usecases/user/usecase.go:52]: like request entity, db entity... etc. 
not now

File: internal/entity/user.go (hunk 5 above)
Comment (igkostyuk) [reply internal/entity/user.go:25]: it wos my mistake to add it,

File: internal/server/http/auth/handlers.go (hunk 4 above)
Comment (glbter) [reply internal/server/http/auth/handlers.go:79]: I thought about dividing user from his password. Auth service will handle it

File: internal/usecases/user/usecase.go (hunk 2 above)
Comment (glbter) [reply internal/usecases/user/usecase.go:38]: for now, we could pass userRequest

File: internal/usecases/user/usecase.go (hunk 3 above)
Comment (igkostyuk) [reply internal/usecases/user/usecase.go:52]: no

File: internal/server/http/auth/handlers.go (hunk 4 above)
Comment (igkostyuk) [reply internal/server/http/auth/handlers.go:79]: ye, password and password hash

Comment (igkostyuk) [reply internal/server/http/auth/handlers.go:79]: i need to divide http layer, it's too mixed with business logic now

File: internal/usecases/user/usecase.go (hunk 2 above)
Comment (igkostyuk) [reply internal/usecases/user/usecase.go:38]: but if we use not http something it will be hard

File: internal/entity/user.go (hunk 5 above)
Comment (igkostyuk) [reply internal/entity/user.go:25]: we should separate http and usecase and usecase should do same for other cases than http

File: internal/server/http/auth/handlers.go (hunk 4 above)
Comment (glbter) [reply internal/server/http/auth/handlers.go:79]: ok, good luck then.

Comment (igkostyuk) [reply internal/server/http/auth/handlers.go:79]: any help ? :D

File: internal/entity/user.go (hunk 5 above)
Comment (glbter) [reply internal/entity/user.go:25]: yep. then we will wait for changes of your second branch. Need to find a boilerplaite generator for mappers

File: internal/server/http/auth/handlers.go (hunk 4 above)
Comment (glbter) [reply internal/server/http/auth/handlers.go:79]: for now i will wait for other branch. then we could make changes separetly. Or you wouldn't touch all code?

Comment (glbter) [reply internal/server/http/auth/handlers.go:79]: if so, I can start

**Review by glbter — APPROVED** [review]
ok to proceed. will refactor later to make working and have better arch

File: internal/server/http/auth/handlers.go (hunk 4 above)
Comment (igkostyuk) [reply internal/server/http/auth/handlers.go:79]: i can do everything that depends on user and leave auth part on you, or any idea ?

File: internal/entity/user.go (hunk 5 above)
Comment (igkostyuk) [reply internal/entity/user.go:25]: but what mappers do we need  ?

File: internal/server/http/auth/handlers.go (hunk 4 above)
Comment (glbter) [reply internal/server/http/auth/handlers.go:79]: it depends on what part you need, if not auth, I will do something with it next week

**Comment by igkostyuk** [comment]
freez till demo

**Pull request closed without merging**